### ISSUE TITLE
Crash in add-import

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#38857]
Getting this callstack that is bringing down add-import:

```
System.InvalidOperationException : This instance must be flushed before being reinitialized.
   at Nerdbank.Streams.BufferTextWriter.Initialize(IBufferWriter`1 bufferWriter,Encoding encoding)
   at StreamJsonRpc.JsonMessageFormatter.WriteJToken(IBufferWriter`1 contentBuffer,JToken json)
   at StreamJsonRpc.JsonMessageFormatter.Serialize(IBufferWriter`1 contentBuffer,JsonRpcMessage message)
   at StreamJsonRpc.HeaderDelimitedMessageHandler.Write(JsonRpcMessage content,CancellationToken cancellationToken)
   at StreamJsonRpc.PipeMessageHandler.WriteCoreAsync(JsonRpcMessage content,CancellationToken cancellationToken)
   at async StreamJsonRpc.MessageHandlerBase.WriteAsync(<Unknown Parameters>)
   at async StreamJsonRpc.JsonRpc.TransmitAsync(<Unknown Parameters>)
   at System.Threading.Tasks.ValueTask.ThrowIfCompletedUnsuccessfully()
   at async StreamJsonRpc.JsonRpc.InvokeCoreAsync[TResult](<Unknown Parameters>)
   at async Microsoft.ServiceHub.Framework.RemoteServiceBroker.GetPipeAsync(<Unknown Parameters>)
   at async Microsoft.ServiceHub.Client.HubClient.GetPipeAsync(<Unknown Parameters>)
   at async Microsoft.ServiceHub.Client.HubClient.RequestServiceAsync(<Unknown Parameters>)
   at async Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.Connections.RetryRemoteCallAsync[TException,TResult](<Unknown Parameters>)
   at async Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.Connections.RequestServiceAsync(<Unknown Parameters>)
   at async Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.ConnectionManager.TryCreateNewConnectionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Remote.RemoteHostClientExtensions.TryCreateSessionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Remote.RemoteHostClientExtensions.TryCreateCodeAnalysisSessionAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Remote.RemoteHostClientExtensions.TryRunCodeAnalysisRemoteAsync[T](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.GetFixesAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.AddImport.AbstractAddImportFeatureService`1.GetFixesForDiagnosticsAsync[TSimpleNameSyntax](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.AddImport.AbstractAddImportCodeFixProvider.RegisterCodeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeFixes.CodeFixService.GetCodeFixesAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions.PerformFunctionAsync[T](<Unknown Parameters>)
```

Something is up with OOP.

**Comment by jmarolf** [comment]
OOPs

**Comment by sharwell** [comment]
Duplicate of https://github.com/microsoft/vs-threading/issues/521

**Comment by jinujoseph** [comment]
we planning to service this fix in 16.4
